src/tree/iterators.rs

```diff
@@ -0,0 +1,252 @@
+        Ancestors {
+            tree: tree,
+            node_id: Some(node_id),
+        }
+    }
+}
+
+impl<'a, T> Iterator for Ancestors<'a, T> {
+    type Item = &'a Node<T>;
+
+    fn next(&mut self) -> Option<&'a Node<T>> {
+        if let Some(current_id) = self.node_id.clone() {
+
+            if let Some(parent_id) = self.tree.get_unsafe(&current_id).parent() {
+
+                let parent = self.tree.get_unsafe(&parent_id);
+
+                self.node_id = Some(parent_id.clone());
+
+                return Some(parent);
+            } else {
+                self.node_id = None;
+            }
+        }
+        None
+    }
+}
+
+///
+/// An Iterator over the ancestors of a `Node`.
+///
+/// Iterates over `NodeId`s instead of over the `Node`s themselves.
+///
+pub struct AncestorIds<'a, T: 'a> {
+    tree: &'a Tree<T>,
+    node_id: Option<NodeId>,
+}
+
+impl<'a, T> IteratorNew<'a, T, AncestorIds<'a, T>> for AncestorIds<'a, T> {
+    fn new(tree: &'a Tree<T>, node_id: NodeId) -> AncestorIds<'a, T> {
+        AncestorIds {
+            tree: tree,
+            node_id: Some(node_id),
+        }
+    }
+}
+
+impl<'a, T> Iterator for AncestorIds<'a, T> {
+    type Item = &'a NodeId;
+
+    fn next(&mut self) -> Option<&'a NodeId> {
+        if let Some(current_id) = self.node_id.clone() {
+            if let Some(parent_id) = self.tree.get_unsafe(&current_id).parent() {
+                self.node_id = Some(parent_id.clone());
+                return Some(parent_id);
+            } else {
+                self.node_id = None;
+            }
+        }
+        None
+    }
+}
+
+///
+/// An Iterator over the children of a `Node`.
+///
+/// Iterates over the child `Node`s of a given `Node` in the `Tree`.  Each call to `next` will
+/// return an immutable reference to the next child `Node`.
+///
+pub struct Children<'a, T: 'a> {
+    tree: &'a Tree<T>,
+    child_ids: Iter<'a, NodeId>,
+}
+
+impl<'a, T> IteratorNew<'a, T, Children<'a, T>> for Children<'a, T> {
+    fn new(tree: &'a Tree<T>, node_id: NodeId) -> Children<'a, T> {
+        Children {
+            tree: tree,
+            child_ids: tree.get_unsafe(&node_id).children().as_slice().iter(),
+        }
+    }
+}
+
+impl<'a, T> Iterator for Children<'a, T> {
+    type Item = &'a Node<T>;
+
+    fn next(&mut self) -> Option<&'a Node<T>> {
+        if let Some(ref next_child_id) = self.child_ids.next() {
+            return Some(self.tree.get_unsafe(next_child_id));
+        }
+        None
+    }
+}
+
+///
+/// An Iterator over the children of a `Node`.
+///
+/// Iterates over `NodeId`s instead of over the `Node`s themselves.
+///
+pub struct ChildrenIds<'a> {
+    child_ids: Iter<'a, NodeId>,
+}
+
+impl<'a, T> IteratorNew<'a, T, ChildrenIds<'a>> for ChildrenIds<'a> {
+    fn new(tree: &'a Tree<T>, node_id: NodeId) -> ChildrenIds<'a> {
+        ChildrenIds { child_ids: tree.get_unsafe(&node_id).children().as_slice().iter() }
+    }
```

Comment:
Yep.
What you really want here is this nightly feature: https://github.com/rust-lang/rfcs/pull/1522

But since there is no real reason, this crate should not work on stable, on second thought, it might be better to hide the underlying Iterator, so you can more easily change it, until RFC 1522 has been stabilized.